test/app-luatest/uri_test.lua

```diff
@@ -0,0 +1,82 @@
+local uri = require("uri")
+local t = require("luatest")
+
+local g = t.group("uri_integration")
+
+local g_opts = t.group("uri_integration_opts", {
+    { name = "RFC3986" },
+    { name = "PATH" },
+    { name = "PATH_PART" },
+    { name = "QUERY" },
+    { name = "QUERY_PART" },
+    { name = "FRAGMENT" },
+    { name = "FORM_URLENCODED" },
+})
+
+local build_ascii_table = function()
```

Comment:
Nit: ASCII is 7 bit (code points 0-127), I would rename it to `build_bytes_table` or something similar.